File: CHUONG_7/7.09_MultiThread/app/src/main/java/com/example/myapplication/resource/SongResponseParser.java
package com.example.myapplication.resource;

import com.example.myapplication.model.SongList;
import com.google.gson.Gson;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;

public class SongResponseParser {
    public SongList parse (String url) throws IOException {
        URL urlConn = new URL(url);
        InputStreamReader inputStreamReader = new InputStreamReader(urlConn.openStream());
        BufferedReader reader = new BufferedReader(inputStreamReader);
        Gson gson = new Gson();
        return gson.fromJson(reader, SongList.class);
    }
}

File: CHUONG_7/7.08_Coroutine/app/src/main/java/com/example/coroutine/resource/SongResponseParser.kt
package com.example.coroutine.resource

import com.example.coroutine.model.SongList
import com.google.gson.Gson
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.URL
import kotlin.jvm.Throws

class SongResponseParser {
    @Throws(IOException::class)
    fun parse(url: String): SongList {
        val musicConn = URL(url)
        val reader = BufferedReader(InputStreamReader(musicConn.openStream()))
        return Gson().fromJson(reader, SongList::class.java)
    }
}